### DISCUSSION TITLE
Calling async actions from an action

### DISCUSSION THREAD

**Discussion by martincalvert** [pmndrs/zustand#955] [Q&A] [answered]
This is a general question to make sure I'm not starting on a bad path with zustand. I currently am using a context for my entire app, so I have some spots where I set state then use a callback to call some other async APIs after a user performs an action. it looks like this in the context:
```js
setState({ someState: true }, async () => {
  Promise.all([this.otherAction1, this.otherAction2])
});
```

and I plan to replace it with this in the store:
```js
set({ someState: true })
await Promise.all([
  get().otherAction1(),
  get().otherAction2()
]);
```

Thank you!

**Comment by dai-shi** [comment, marked as answer]
I don't see any issue so far.
